src/core/config/mac.h

```diff
@@ -35,6 +35,7 @@
 #ifndef CONFIG_MAC_H_
 #define CONFIG_MAC_H_
 
+#include "config/device_type_check.h"
```

Comment:
Is it still necessary after we have included "config/device_type_check.h" at the beginning of `openthread-core-config.h`